pkg/proxy/node.go

```diff
@@ -28,41 +28,23 @@ import (
 	utilnode "k8s.io/kubernetes/pkg/util/node"
 )
 
-// NodeEligibleHandler handles the life cycle of the Node's eligibility, as
-// determined by the health server for directing load balancer traffic.
-type NodeEligibleHandler struct {
-	HealthServer *healthcheck.ProxyHealthServer
-}
-
-var _ config.NodeHandler = &NodeEligibleHandler{}
-
-// OnNodeAdd is a handler for Node creates.
-func (n *NodeEligibleHandler) OnNodeAdd(node *v1.Node) { n.HealthServer.SyncNode(node) }
-
-// OnNodeUpdate is a handler for Node updates.
-func (n *NodeEligibleHandler) OnNodeUpdate(_, node *v1.Node) { n.HealthServer.SyncNode(node) }
-
-// OnNodeDelete is a handler for Node deletes.
-func (n *NodeEligibleHandler) OnNodeDelete(node *v1.Node) { n.HealthServer.SyncNode(node) }
-
-// OnNodeSynced is a handler for Node syncs.
-func (n *NodeEligibleHandler) OnNodeSynced() {}
-
 // NodeManager handles the life cycle of kube-proxy based on the NodeIPs and PodCIDRs.
 // Implements the config.NodeHandler interface
 // (ref: https://issues.k8s.io/111321)
 type NodeManager struct {
 	nodeIPs           []net.IP
 	podCIDRs          []string
 	localModeNodeCIDR bool
+	healthServer      *healthcheck.ProxyHealthServer
 	logger            klog.Logger
 }
 
-func NewNodeManager(ctx context.Context, nodeIPs []net.IP, podCIDRs []string, localModeNodeCIDR bool) *NodeManager {
+func NewNodeManager(ctx context.Context, nodeIPs []net.IP, podCIDRs []string, localModeNodeCIDR bool, healthServer *healthcheck.ProxyHealthServer) *NodeManager {
```

Comment:
When I was thinking about this, I wasn't sure if it would make more sense for `NodeManager` to take a `ProxyHealthServer` argument, or for `ProxyHealthServer` to take a `NodeManager` argument.

In particular, note that `ProxyHealthServer` doesn't actually react immediately to `SyncNode`; it doesn't actually look at the node object until the next time it constructs a health check response, so in theory, it could just call `nodeManager.GetCurrentNode()` at that point.

(But again, I didn't actually decide for sure that this made sense, I just wondered if it did.)